File: Solutions/2149. Rearrange Array Elements by Sign/2149.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int posInd = getNextPos(nums, 0);
        int negInd = getNextNeg(nums, 0);
        int[] ans = new int[nums.length];
        int ansInd = 0;
        while(posInd != -1 && negInd != -1){
            ans[ansInd] = nums[posInd];
            ansInd++;
            posInd = getNextPos(nums, posInd+1);
            ans[ansInd] = nums[negInd];
            ansInd++;
            negInd = getNextNeg(nums, negInd+1);
        }
        return ans;
    }
    public int getNextPos(int[] nums, int ind){
        for(int i=ind;i<nums.length;i++){
            if (nums[i] > 0){
                return i;
            }
        }
        return -1;
    }
    public int getNextNeg(int[] nums, int ind){
        for(int i=ind;i<nums.length;i++){
            if (nums[i] < 0){
                return i;
            }
        }
        return -1;
    }
}

File: Solutions/2149. Rearrange Array Elements by Sign/2149.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:

        pos_index = -1
        neg_index = -1

        def findNextPos(index):
            for i in range(index, len(nums)):
                if nums[i] > 0:
                    return i
            return -1

        def findNextNeg(index):
            for i in range(index, len(nums)):
                if nums[i] < 0:
                    return i
            return -1

        pos_index, neg_index = findNextPos(0), findNextNeg(0)

        ans = []

        while pos_index != -1 and neg_index != -1:
            ans.append(nums[pos_index])
            pos_index = findNextPos( pos_index + 1 )
            ans.append(nums[neg_index])
            neg_index = findNextNeg( neg_index + 1 )

        return ans

